apps/google-analytics-4/frontend/src/components/config-screen/api-access/ApiAccessPage.tsx

```diff
@@ -1,58 +1,71 @@
-import React from 'react';
-import InstalledServiceAccountCard from './service-account/InstalledServiceAccountCard';
-import SetupServiceAccountCard from './service-account/SetupServiceAccountCard';
+import React, { useState } from 'react';
+import DisplayServiceAccountCard from 'components/config-screen/api-access/display/DisplayServiceAccountCard';
+import SetupServiceAccountCard from 'components/config-screen/api-access/setup/SetupServiceAccountCard';
 import { Subheading, Paragraph, Stack } from '@contentful/f36-components';
-import { ServiceAccountKeyId, ServiceAccountKey } from 'types';
+import useKeyService from 'hooks/useKeyService';
 
 interface Props {
-  isValid: boolean;
-  errorMessage: string;
-  isRequired: boolean;
-  currentServiceAccountKeyId: ServiceAccountKeyId | null;
-  currentServiceAccountKey: ServiceAccountKey | null;
-  serviceAccountKeyFile: string;
-  onKeyFileChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
-  isInEditMode: boolean;
-  onEditGoogleAccountDetails: React.MouseEventHandler<HTMLButtonElement>;
-  onCancelGoogleAccountDetails: React.MouseEventHandler<HTMLButtonElement>;
+  onAccountSummariesChange: Function;
```

Comment:
What do you thin about making this optional so we don't have to declare empty functions so often:
```suggestion
  onAccountSummariesChange?: Function;
```

Reply:
Pretty sure there would be cascading changes from this, but it might clean things up a bit